web/web.go

```diff
@@ -526,38 +529,193 @@ func dropForeignKey(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	//check new name for non usage before in another table, foreign key constraint or index
+	if status, err := canRename(newNames, table); !status {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
 	}
+
 	sp := sessionState.conv.SpSchema[table]
-	position, err := strconv.Atoi(pos)
+
+	// Update session with renamed foreignkeys
+	newFKs := []ddl.Foreignkey{}
+	for _, foreignKey := range sp.Fks {
+		if newName, ok := changeMap[foreignKey.Name]; ok {
+			foreignKey.Name = newName
+		}
+		newFKs = append(newFKs, foreignKey)
+	}
+	sp.Fks = newFKs
+
+	sessionState.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sessionState.conv)
+}
+
+func renameIndexes(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	reqBody, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("Error converting position to integer"), http.StatusBadRequest)
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+	}
+
+	changeMap := map[string]string{}
+	if err = json.Unmarshal(reqBody, &changeMap); err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
 		return
 	}
-	if position < 0 || position >= len(sp.Fks) {
-		http.Error(w, fmt.Sprintf("No foreign key found at position %d", position), http.StatusBadRequest)
+	//check new name for spanner name validity
+	newNames := getValuesAsLowerCaseSlice(changeMap)
+	if validity, invalidNewNames := checkSpannerNamesValidity(newNames); !validity {
+		http.Error(w, fmt.Sprintf("New name validation fails for following : %v", invalidNewNames), http.StatusBadRequest)
+		return
+	}
+
+	//check new name for non usage before in another table, foreign key constraint or index
+	if status, err := canRename(newNames, table); !status {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	sp := sessionState.conv.SpSchema[table]
+
+	// Update session with renamed secondary indexes
+	newIndexes := []ddl.CreateIndex{}
+	for _, index := range sp.Indexes {
+		if newName, ok := changeMap[index.Name]; ok {
+			index.Name = newName
+		}
+		newIndexes = append(newIndexes, index)
+	}
+	sp.Indexes = newIndexes
+
+	sessionState.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sessionState.conv)
+}
+func addIndexes(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+	}
+
+	newIndexes := []ddl.CreateIndex{}
+	if err = json.Unmarshal(reqBody, &newIndexes); err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	//check new name for spanner name validity
+	newNames := getNameSlice(newIndexes)
```

Comment:
Similar comment as `getValuesAsLowerCaseSlice`, you can just inline the for loop here instead of declaring a new function `getNameSlice`.